include/giskard_core/yaml_parser.hpp

```diff
@@ -1596,6 +1596,49 @@ namespace YAML {
     }
   };
 
+  //
+  // parsing of alias reference specs
+  //
+
+  inline bool is_alias_reference(const Node& node)
+  {
+    if(!node.IsScalar() || is_const_double(node))
```

Comment:
I think you're right. This shouldn't work. Funnily, I kept on "programming in my head" while I was on the bus. At some point, I had this intuition that this condition should be wrong. Interesting that you also spotted it. I'll have a look at it.